### DISCUSSION TITLE
After enabling “DO_NOT_TRANSCODE_VIDEO” uploaded video’s do not play but the thumbnail works

### DISCUSSION THREAD

**Discussion by foeyonghai** [mediacms-io/mediacms#1258] [Q&A]
**Describe the issue**
I downloaded and created multiple instances of MediaCMS on an updated Ubuntu server and enabled the “DO_NOT_TRANSCODE_VIDEO” option that was implemented not too long ago. After uploading videos it shows a thumbnail but after clicking on the thumbnail and getting on the page of the video itself it doesn’t play and only shows a black image at the place where the player is expected.

**To Reproduce**
Steps to reproduce the issue:

1. Download MediaCMS
2. Tweak the /deploy/settings.py file and change “DO_NOT_ENCODE_VIDEO” to enabled
3. Installing an instance of the newest MediaCMS using the regular docker-compose file, so docker-compose.yml. I did not tweak any settings except for the password of the admin account of MediaCMS itself.
4. Upload a video, I tried .vob, .mp4 and some others (just the regular files)
5. Try to play the video

**Expected behavior**
A working video

**Screenshots**
Will provide but I’m not at home now

**Environment (please complete the following information):**
Ubuntu Server, updated
Docker and docker compose

**Comment by foeyonghai** [comment]
I am not at home now so some places like /deploy/cms/settings.py I need to double check, doing this relying on my memories now.

**Comment by foeyonghai** [comment]
Okay I did some extended tests and it gets clearer

- .vob files are not working, thumbnails DO work
- .mkv files are not working, thumbnails DO work
- .avi files are not working, thumbnails DO work
- **.mp4 files DO work**, thumbnails DO work

The behaviour above is in 2 seperate tenants.

I am wondering at this point: which  video extentions are supported and which not?

**Comment by TheFlipside** [comment]
I had a similar issue but I'm using a local installation.
I found that when i set "DO_NOT_ENCODE_VIDEO = True" in /home/mediacms.io/mediacms/cms/local_settings.py even after restarting the mediacms service I still had the system do encodings for newly uploaded media.
The setting had to be set in /home/mediacms.io/mediacms/cms/settings.py and now it works, newly uploaded media do not trigger any encoding and the only selectable quality on a media is "0-original".
It seems the setting is not taken in effect when put in the local settings.

**Comment by shootie22** [comment]
@foeyonghai have you ever managed to get .mkv files to play? I'm currently running into the same issue :(

**Comment by thecoffeebin** [comment]
Wondering the same. Is MKV supported or any workaround to make it play?

Because with the latest version of MediaCMS I can upload MKV video files but I can only listen to audio playing (and no video) on the web.

**Comment by mgogoulos** [comment]
It seems that only a few of the video files are playable by videojs - definitely mp4, some others too, but there are file types that are not playable
